fix(openstack_blockstorage_volume_v3) : Allow image_id to be imported when other ids are not present

jsittaro opened the pull request:
This should fix the problem with image_id that was not imported into the state by adding it in the importing process. But this comes with two catches:

- The first one happens when image_id and at other id is present at the same time because of OpenStack keeping ``ImageMetadata`` when using a snapshot or another volume as base, therefore making a conflict who the two ids. To make sure that they are not conflicting each other, image_id is only set if the other ids are not present.
- As of today, OpenStack are accepting an image name on image_id field as there are resolving it internally[1]. But in this scenario, as the import function imports the ID and not the name in ``image_id`` field, Terraform will try to force-recreate the volume when applying. For a workaround, user should be using directly a image_id or get it from ``openstack_images_image_v2`` datasource.

As this is my first contribution, I am not sure if this is the correct solution, so I am open to comment if this PR needs improvement.

Fixes #1265

[1]: https://github.com/openstack/cinder/blob/635bdab2c0d44e95e499a1d13296446171d5e4ef/cinder/api/v3/volumes.py#L364 and https://github.com/openstack/cinder/blob/635bdab2c0d44e95e499a1d13296446171d5e4ef/cinder/api/v2/volumes.py#L136

kayrus commented on openstack/resource_openstack_blockstorage_volume_v3.go:
```diff
@@ -283,6 +283,16 @@ func resourceBlockStorageVolumeV3Read(ctx context.Context, d *schema.ResourceDat
+	_, sourcevolExists := d.GetOk("source_vol_id")
+
+	if !snapshotExists && !backupExists && !sourcevolExists {
+		d.Set("image_id", v.VolumeImageMetadata["image_id"])
```
does it makes sense always set this attribute without any condition?

jsittaro replied:
Hello @kayrus and thanks for the review. Also, sorry for the delay of the response, I was a bit occupied at my work.

If I understand your question, you ask why not set ``image_id`` all time without any condition. If this is the question, this is not possible for one reason: ``image_id``, ``snapshot_id``, ``backup_id`` and ``source_vol_id`` is excluded each other (``ConflictWith`` statement on the schema) and two ids may be present both in some circumstance.
For example, when you make a snapshot of a volume that was originally created with an image and someone decides to create a new volume with that snapshot, ``ImageMetadataVolume`` is carried over to the new volume and we got ``image_id`` and ``snapshot_id`` as the same time.
In this case:
    - Terraform will try to replace my volume because image_id is not set in my configuration and the plan wants to remove it therefore forcing recreation of the volume.
    - If I add ``image_id`` to match the volume on OpenStack and the state on Terraform, Terraform will disallow me to do that because now I have two values that are conflicting each other.
So, due to that, ``image_id`` need to be behind a condition to make sure that image ID is out of way when Terraform importing or refreshing it into the state when the volume is based on a snapshot, backup or a source volume.

kayrus replied:
I guess terraform will recreate the resource even with the condition.
I suppose you need this for an import only, right? Then let's adjust the import function only and leave the read function as is.

jsittaro commented:
Hello,
I have moved this into the import function only and changed the description to match the current state of this PR. 
On the way, I added a import test to ensure that image_id is present when it should be here and slightly changed the condition to pass the basic import test as there were falling when image_id was present with empty string.

kayrus reviewed (approved):
LGTM, thanks!
